### ISSUE TITLE
[HR] Enable WASD movement

### ISSUE THREAD

**Issue by afkvido** [Prodigy-Hacking/ProdigyMathGameHacking#1310]
# Enable WASD movement
Requested by [afkvido](https://github.com/afkvido)

## Describe your hack:
A toggle that would let the character move with WASD

## (Optional) Additional information
Idea:
- variable `X` = [Get player location X]
- variable `Y` = [Get player location Y]
.
- When W key pressed: [Set player location to x:(`X`) y:(`Y` + 4)

## Have you made sure this hack isn't available yet?* (Yes/No)
The hack is not available in Will's Cheat Menu.

**Comment by donaldli2020** [comment]
good idea

**Comment by donaldli2020** [comment]
```
window.onkeydown = function() {	
	var keyCode = event.keyCode;
	
	if (keyCode == 87) {	
		// change y by 4
	}
	if (keyCode == 65) {	
		// change x by -4
	}
	if (keyCode == 83) {	
		// change y by -4
	}
	if (keyCode == 68) {	
		// change x by 4
	}
}
```

**Comment by afkvido** [comment]
looks quite nice! we just need to implement this into WCM!

**Comment by LeoBadeaux** [comment]
Í've wanted to be able to do this for a while.

> looks quite nice! we just need to implement this into WCM!

I´ll implement this later!

**Comment by 32n** [comment]
This is actually a wonderful idea. Moving with a mouse is so janky.

**Comment by LeoBadeaux** [comment]
I believe in order for us to do this we have to do some physics calculations to determine the move location based on the players speed. In addition, it would be nice to have this automatically enabled.

**Comment by afkvido** [comment]
> This is actually a wonderful idea. Moving with a mouse is so janky.

After playing minecraft and other competitive wasd games for like a year and returning to prodigy, it's quite impossible to move..

**Comment by afkvido** [comment]
> I believe in order for us to do this we have to do some physics calculations to determine the move location based on the players speed. In addition, it would be nice to have this automatically enabled.

+upvote

**Comment by 32n** [comment]
Wait wait... VIM CONTROLS! hjkl to move! It's way better than WASD for 2d movement!

**Comment by ArcerionDev** [comment]
> Wait wait... VIM CONTROLS! hjkl to move! It's way better than WASD for 2d movement!

no

**Comment by 32n** [comment]
> I believe in order for us to do this we have to do some physics calculations to determine the move location based on the players speed. In addition, it would be nice to have this automatically enabled.

no this might be unnecessary. Just make it click to the side of the player, if possible.

**Comment by 32n** [comment]
> > Wait wait... VIM CONTROLS! hjkl to move! It's way better than WASD for 2d movement!
> 
> no

hmph! mouse navigator! why use WASD when you can keep your hand firmly on the home row with vim navigation?

**Comment by LeoBadeaux** [comment]
> > > Wait wait... VIM CONTROLS! hjkl to move! It's way better than WASD for 2d movement!
> > 
> > 
> > no
> 
> hmph! mouse navigator! why use WASD when you can keep your hand firmly on the home row with vim navigation?

no! even better! what if we made a hack that lets you walk by clicking where you want to walk to?

**Comment by 32n** [comment]
I don't see why it can't co-exist with the vim controls :^)

your sarcasm does little to disprove my point!

**Comment by afkvido** [comment]
> I don't see why it can't co-exist with the vim controls :^)
> 
> your sarcasm does little to disprove my point!

WASD is better

**Comment by 32n** [comment]
> > I don't see why it can't co-exist with the vim controls :^)
> > your sarcasm does little to disprove my point!
> 
> WASD is better

How so? Your hand is off the home row. hjkl makes sense for any vim user.

**Comment by afkvido** [comment]
EVERYONE USES WASD bruh

**Comment by 32n** [comment]
...not non-gamers who use vim.

**Comment by LeoBadeaux** [comment]
> ...not non-gamers who use vim.

vim is for losers

**Comment by 32n** [comment]
vim is for people who want a frictionless workflow!

**Comment by LeoBadeaux** [comment]
> vim is for people who want a frictionless workflow!

vim is for programmers who live in their moms basement with a 2003 core i5 windows machine with a dark cloak on and no lights on.
![image](https://user-images.githubusercontent.com/79180322/152020877-a2e4d08f-e537-48cd-ac28-15fe4ae8f719.png)

**Comment by 32n** [comment]
sounds like someone had trouble learning the shortcuts.

**Comment by LeoBadeaux** [comment]
> sounds like someone had trouble learning the shortcuts.

Your location is a lie because there is no way you have a Great Neck looking down at your book of vim shortcuts every 2 seconds. You probably write code with spaces too.

**Comment by afkvido** [comment]
32n, nobody uses Vim because i only learned about it today

**Comment by 32n** [comment]
> 32n, nobody uses Vim because i only learned about it today

Any decently experienced programmer knows what vim is.

**Comment by 32n** [comment]
> > sounds like someone had trouble learning the shortcuts.
> 
> Your location is a lie because there is no way you have a Great Neck looking down at your book of vim shortcuts every 2 seconds. You probably write code with spaces too.

sounds like someone had trouble learning the shortcuts. 😬

**Comment by afkvido** [comment]
ur bad

**Comment by LeoBadeaux** [comment]
> > > sounds like someone had trouble learning the shortcuts.
> > 
> > 
> > Your location is a lie because there is no way you have a Great Neck looking down at your book of vim shortcuts every 2 seconds. You probably write code with spaces too.
> 
> sounds like someone had trouble learning the shortcuts. 😬

you already used that joke you bozo

**Comment by ArcerionDev** [comment]
![](https://cdn.discordapp.com/emojis/754153286575063042.webp)

**Comment by UntrustableRus** [comment]
here's a really lazy implementation that kinda works but not great
```js
document.addEventListener("keydown", function (event) {
	switch (event.key) {
		case "w":
			_.player._playerContainer.y = _.player._playerContainer.y - 1;
			break;
		case "s":
			_.player._playerContainer.y = _.player._playerContainer.y + 1;
			break;
		case "a":
			_.player._playerContainer.x = _.player._playerContainer.x - 1;
			break;
		case "d":
			_.player._playerContainer.x = _.player._playerContainer.x + 1;
			break;
	}
});
```

**Comment by afkvido** [comment]
i like that

**Comment by donaldli2020** [comment]
I would use 10
```
document.addEventListener("keydown", function (event) {
	switch (event.key) {
		case "w":
			_.player._playerContainer.y -= 10;
			break;
		case "s":
			_.player._playerContainer.y += 10;
			break;
		case "a":
			_.player._playerContainer.x -= 10;
			break;
		case "d":
			_.player._playerContainer.x += 10;
			break;
	}
});
```

**Comment by afkvido** [comment]
sure!

**Comment by LeoBadeaux** [comment]
> here's a really lazy implementation that kinda works but not great
> 
> ```js
> document.addEventListener("keydown", function (event) {
> 	switch (event.key) {
> 		case "w":
> 			_.player._playerContainer.y = _.player._playerContainer.y - 1;
> 			break;
> 		case "s":
> 			_.player._playerContainer.y = _.player._playerContainer.y + 1;
> 			break;
> 		case "a":
> 			_.player._playerContainer.x = _.player._playerContainer.x - 1;
> 			break;
> 		case "d":
> 			_.player._playerContainer.x = _.player._playerContainer.x + 1;
> 			break;
> 	}
> });
> ```

What I mean by it only half-working is that there is no walking animation, and if you walk up to a shop display or a wheel display or something that would display a modal in general, It won't display, you would have to click walk to it. This does not mean that you can't go through different zones, that works fine. In addition, I haven't tested this to see if it will broadcast your new position to other people playing.

**Comment by afkvido** [comment]
True, but some WASD movement is better than none

**Comment by LeoBadeaux** [comment]
> True, but some WASD movement is better than none

of course, but this feature could use some immense more polishing.

**Comment by afkvido** [comment]
Definetly! :)

**Comment by afkvido** [comment]
@LeoBadeaux Wait, does this mean we can phase through solid ground? If so, then HELL YEA

**Comment by LeoBadeaux** [comment]
> @LeoBadeaux Wait, does this mean we can phase through solid ground? If so, then HELL YEA

Prodigy is 2D, therefore you can't go through a 3D Plane. This just means we can walk on stores and barriers and stuff.

**Comment by afkvido** [comment]
yay!!!!

**Comment by PatheticMustan** [comment]
??? Don't we already have a "remove barriers" hack...? Maybe it wasn't implemented yet but I believe in prodigy.world it should hold a 2d array of 1s and 0s that determines if you can walk to certain areas or not. If you just patch that to be all 1s it should be the same thing

**Comment by LeoBadeaux** [comment]
> ??? Don't we already have a "remove barriers" hack...? Maybe it wasn't implemented yet but I believe in prodigy.world it should hold a 2d array of 1s and 0s that determines if you can walk to certain areas or not. If you just patch that to be all 1s it should be the same thing

Please read the issue title.

**Comment by PatheticMustan** [comment]
I wasn't replying to the issue, I was replying to @afkvido's question
> "does this mean we can phase through solid ground"

I assure you I have indeed read the issue title! yeah wow reading comprehension woooo

**Comment by afkvido** [comment]
reading comparehernsion yes how hack prodigy pls i want coin and flex on grilzz

`Sent From My iPhone ™️`

**Comment by LeoBadeaux** [comment]
Unassigned because I am not wanted here.

**Comment by PatheticMustan** [comment]
passy aggy..?
